packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -1034,44 +1040,101 @@ describe("Teal Opcodes", function () {
 
 	describe("Ed25519verify", function () {
 		const stack = new Stack<StackElem>();
+		let interpreter = new Interpreter();
+		let elonAcc: AccountStoreI;
+		const elonPk = decodeAddress(elonAddr).publicKey;
 
-		it("should push 1 to stack if signature is valid", function () {
-			const account = generateAccount();
-			const toSign = new Uint8Array(Buffer.from([1, 9, 25, 49]));
-			const signed = signBytes(toSign, account.sk);
-
-			stack.push(toSign); // data
-			stack.push(signed); // signature
-			stack.push(decodeAddress(account.addr).publicKey); // pk
+		const resetInterpreterState = (): void => {
```

Comment:
You do not need to reset the whole `interpreter` since you only need `interpreter.program`

Reply:
we actually need to reset it. Otherwise we will have an error due to the stack length

Reply:
what error does it throw?

Reply:
`ASSERT_STACK_LENGTH`

Reply:
You are right, I was able to get rid of majority of the function. We didn't need that. Thanks